makefiles/stdio.inc.mk

```diff
@@ -15,6 +15,7 @@ ifneq (,$(filter newlib picolibc,$(USEMODULE)))
 endif
 
 ifneq (,$(filter stdio_cdc_acm,$(USEMODULE)))
+  USEMODULE += picolibc_stdout_buffered
```

Comment:
Shouldn't these be only if `picolibc` is used?

Reply:
Since it's a pseudo-module it will do no harm.

Reply:
We could also turn it into a generic `stdout_buffering` pseudo-module.

Reply:
It somewhow does, since if I do `info-modules` I see something that is not used and would see maybe `newlib` and `picolibc` which would be weird. IMO its better to guard them.

Reply:
And what about

```
USEMODULE += stdout_buffered
```

instead of

```Make
ifneq (,$(filter picolibc,$(USEMODULE)))
  USEMODULE += picolibc_stdout_buffered
endif
```

Reply:
> IIRC on newlib stdout is buffered by default.

If that is the case then yes your proposal makes sense, is there somehwere to reference this statement ?

Reply:
I remember hearing about it in @keith-packard's [picolibc talk](https://www.youtube.com/watch?v=SC6aBezNFFQ).
I also found [this bug report](https://answers.launchpad.net/gcc-arm-embedded/+question/246038) complaining that buffering can't be turned off with `nano.specs`.

Reply:
picolibc doesn't have any internal buffering for stdin/stdout/stderr. When you use fopen/fdopen, it does because those are backed by the POSIX apis (open/close read/write, etc) and have a buffered implementation inside picolibc's POSIX file backend.

Reply:
> picolibc doesn't have any internal buffering for stdin/stdout/stderr. 

Yes but IIRC newlib does - so enabling `stdout_buffered` would simply be the default there, whereas with picolibc it would enable the stdout buffering you added to the RIOT glue code.